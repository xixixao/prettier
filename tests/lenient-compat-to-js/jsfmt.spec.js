function parser(text, { babylon }) {
  const plugins = ["lenient", "lenientCompat", "flow", "jsx"];
  return babylon(text, {}, {}, { plugins });
}

run_spec(__dirname, [parser], {
  bracketSpacing: false,
  semi: true
});
