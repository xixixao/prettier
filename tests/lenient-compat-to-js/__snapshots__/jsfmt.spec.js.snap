// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrows.js 1`] = `
const f = x => {
  x * square(x);
}

const g = x =>
  x * square(x);

const h = () => {}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const f = x => {
  x * square(x);
};

const g = x => {
  x * square(x);
};

const h = () => {};

`;

exports[`conditions.js 1`] = `

if aouhasduosahdouasdhsaoudhsaduosahdousahdsaoudhsaoduh &&
    asdhousadhsaoudhsoaudhsaoudh
  f()
else if aouhasduosahdouasdhsaoudhsaduosahdousahdsaoudhsaoduh &&
    asdhousadhsaoudhsoaudhsaoudh
  g()

if f(
      aouhasduosahdouasdhsaoudhsaduosahdousahdsaoudhsaoduh &&
        asdhousadhsaoudhsoaudhsaoudh
    )
  f()
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
if (
  aouhasduosahdouasdhsaoudhsaduosahdousahdsaoudhsaoduh &&
  asdhousadhsaoudhsoaudhsaoudh
) {
  f();
} else if (
  aouhasduosahdouasdhsaoudhsaduosahdousahdsaoudhsaoduh &&
  asdhousadhsaoudhsoaudhsaoudh
) {
  g();
}

if (
  f(
    aouhasduosahdouasdhsaoudhsaduosahdousahdsaoudhsaoduh &&
      asdhousadhsaoudhsoaudhsaoudh
  )
) {
  f();
}

`;

exports[`declarations.js 1`] = `
const x = 3;
let n = 0;
n = 1;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const x = 3;
let n = 0;
n = 1;

`;
